mono/mini/debugger-agent.c

```diff
@@ -2882,9 +2886,11 @@ resume_vm (void)
 	//g_assert (err == 0);
 
 	if (suspend_count == 0)
-		mono_threadpool_resume ();
-
+		tp_resume = TRUE;
 	mono_loader_unlock ();
+
+	if (tp_resume)
+		mono_threadpool_resume ();
```

Comment:
I'm not super familiar with the threadpool code, but I think that it does not need the loader lock to be held.

Reply:
No, it shouldn't. The threadpool overall doesn't rely on any runtime specificities, and it could completely be in managed if it wasn't for `static` not being cross-domain.